### DISCUSSION TITLE
Duración de los croutons/snackbar con acción

### DISCUSSION THREAD

**Discussion by PatriRoman** [Telefonica/mistica-design#1929] [Proposals]
### Goal

Reducir el tiempo que aparece en pantalla los croutons con acción.

### Context

Desde O2UK nos levantan que el tiempo que aparece los croutons con acción en pantalla es demasiado tiempo, sobre todo cuando esta acción no es "close" si no que es por ejemplo "undo". Más en específico el crouton cuando estás en tu listado de favoritos y eliminas un item de la lista, que sale un mensaje de éxito de "Removed from favourites" y la opción de "undo".
Se quejan de que está demasiado tiempo en pantalla además en una posición estratégica como es la parte inferior.
Adjunto ejemplos específicos: https://www.figma.com/design/LHob8XrzUxozcET7i9shTz/UX-Flow-Wishlist-Test?node-id=4259-9489&t=o2w6ga1dtt9BuThk-1 

### Impact

El impacto sería la reducción de este tiempo, ya que 10 segundos para la cantidad de texto que puede albergar un crouton de manera cotidiana puede llegar a ser elevado (?)

### Discussion Questions

¿Habría posibilidad de reducir este tiempo? ¿O de tener la posibilidad por ejemplo de elegir entre crouton de duración más elevada y menos elevada, es decir, elegir entre varios slots de tiempo? Siempre respetando la accesibilidad claro.

**Comment by yceballost** [comment]
~Hello!~

~Veo varios caminos.~
~1. Una opción sencilla sería añadir la posibilidad de pasarle "short" como duración y que pueda durar 5s. El por defecto se mantendría en 10s cuando lleva acción pero con la posibilidad de sobrescribirlo~
~2. Abrir la posibilidad de pasarle cualquier tiempo, aunque no estoy tan seguro de que esto sea un buen camino por evitar duraciones muy inconsistentes en cada pantalla~

~Yo me iría por el camino 1 por evitar dar esa flexibilidad cuando realmente no creo que sea super decisivo usar valores entre 6-9s.. con dar 5s para "short" y 10s para "long" debería ser suficiente imo.~

**Comment by AnaMontes11** [comment]
~Hola, @PatriRoman ahora mismo como solución a tu necesidad en el que tu crouton tiene acción (undo) si el usuario no quiere esperar esos 10sg permitimos añadir adicionalmente un “close”.~

~Igualmente, creo que podría ser útil incluir una opción “short” (3-5 sg) para mensajes breves y no críticos (pero no llevaría acción).~

**Comment by aweell** [comment]
Tras discutirlo offline parece que el problema de raíz es que actualmente el componente crouton de iOS está tapando la tabBar, por lo que un usuario debe esperar el tiempo definido (10s) antes de interactuar con el componente que queda oculto debajo.

Idealmente debería utilizarse una snackbar en vez de un crouton en ese caso para evitar que exista este conflicto.

**Reply by ieduardogf** [reply]
Eso va a existir igualmente, se presente una vista por arriba que por abajo la posibilidad de que tape la navegación va a estar ahi. Aqui lo importante es saber si es un bug (ya que esta tapando la tabbar) o no, y es una sugerencia de mejora de tiempos o cambios de tiempos por otra razón.

Si esta tapando la tabbar se considera bug.

**Reply by PatriRoman** [reply]
Hola @ieduardogf desde la OB nos dicen "stays up for too long and covers the main navbar. We would like to reduce the length of time so that the main nav bar is not covered for so long".
De lo que entendemos que efectivamente está cubriendo la nav bar, ergo es un bug. Idealmente no deberíamos de cubrir en ningún caso la navegación principal de la app.

**Reply by ieduardogf** [reply]
Creo que entonces va a estar más relacionado con:
- inicialmente esa pantalla no tiene navbar. Se presenta el feedback que se mantiene en pantalla 10 segundos. No cubre nada
- el usuario empieza a navegar por la App y llega durante esos 10 segundos a una pantalla en este caso con navbar. Pero que podría ser con un botón. También quedaría oculto durante el tiempo restante del Crouton

Falta algún vídeo para corroborar esta explicación. Pero aquí entonces habría que decidir si bajamos la duración de los feedbacks en pantalla, en este caso esos 10 segundos o directamente si el usuario navega los retiramos completamente hayan pasado los 10 segundos o no

**Reply by PatriRoman** [reply]
Perdona @ieduardogf que no vi esto, pero hay dos casos en el que la pantalla en donde sale el feedback sí tienen tab bar: https://www.figma.com/design/LHob8XrzUxozcET7i9shTz/UX-Flow-Wishlist-Test?node-id=4259-9489&t=AZrvg9S7eUlKvFl0-1 
Entonces aquí necesitaríamos arreglar o bien bajando la duración del crouton o idealmente que en ningún caso tape la tab bar.

**Comment by aweell** [comment]
This is solved in https://jira.tid.es/browse/IOS-10707
